### ISSUE TITLE
color/substitute issue introduced in the latest patch.

### ISSUE THREAD

**Issue by jrounds** [bokeh/rbokeh#148]
Don't have a time at the moment to write a stand-alone demo of the issue because I need a work-around for tonight, but I can hint at it.

The context is a color vector in a call to ly_crect.  The code ends up in this stack:
```
Available environments had calls:
1: p %>% ly_heatmap(x = Group, y = Peptide, value = value, data = data, colorQuantileFn = ecdf(data$va
2: withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
3: eval(quote(`_fseq`(`_lhs`)), env, env)
4: eval(expr, envir, enclos)
5: `_fseq`(`_lhs`)
6: freduce(value, `_function_list`)
7: withVisible(function_list[[k]](value))
8: function_list[[k]](value)
9: ly_heatmap(., x = Group, y = Peptide, value = value, data = data, colorQuantileFn = ecdf(data$value
10: ly_heatmap.R#18: do.call(ly_crect, call.args)
11: (function (fig, x, y = NULL, data = figure_data(fig), width = 1, height = 1, angle = 0, dilate = FA
12: sub_names(fig, data, grab(x, y, width, height, angle, dilate, color, alpha, hover, url, lname, lgro
13: lapply(arg_obj, parse_values)
14: FUN(X[[i]], ...)
15: lapply(seq_along(x), function(arg_pos) {
    arg_name <- x_names[arg_pos]
    arg_val <- x[[arg_pos]]

16: FUN(X[[i]], ...)
17: sub_fn(arg_val)
18: b_eval_get_symbol(x)
19: eval(parse(text = paste0("substitute(", x$expr, ", x$env)")))
20: parse(text = paste0("substitute(", x$expr, ", x$env)"))
```

With an error of
```
error in parse(text = paste0("substitute(", x$expr, ", x$env)")) : 
  <text>:5:1: unexpected symbol
4: substitute(grey, x$env)
5: substitute
```

grey is not the issue.  I traced in and the text argument of parse evalutes to 
```
rowse[3]> paste0("substitute(", x$expr, ", x$env)")
 [1] "substitute(grey, x$env)"    "substitute(#FFA4A4, x$env)" "substitute(#FFAFAF, x$env)"
 [4] "substitute(grey, x$env)"    "substitute(grey, x$env)"    "substitute(#9A9AFF, x$env)"
 [7] "substitute(#FF6161, x$env)" "substitute(grey, x$env)"    "substitute(grey, x$env)"   
[10] "substitute(#FFD8D8, x$env)" "substitute(#FFF2F2, x$env)" "substitute(#FF4848, x$env)"
[13] "substitute(#3D3DFF, x$env)" "substitute(grey, x$env)"    "substitute(#A4A4FF, x$env)"
[16] "substitute(#4D4DFF, x$env)" "substitute(#1919FF, x$env)" "substitute(#FFBEBE, x$env)"
[19] "substitute(grey, x$env)"    "substitute(#6767FF, x$env)" "substitute(#8080FF, x$env)"
[22] "substitute(grey, x$env)"    "substitute(grey, x$env)"    "substitute(grey, x$env)"   
[25] "substitute(#BEBEFF, x$env)" "substitute(grey, x$env)"    "substitute(grey, x$env)"   
[28] "substitute(grey, x$env)"    "substitute(grey, x$env)"    "substitute(grey, x$env)"   
[31] "substitute(#0A0AFF, x$env)" "substitute(grey, x$env)"    "substitute(#7171FF, x$env)"
[34] "substitute(#B4B4FF, x$env)" "substitute(#E7E7FF, x$env)" "substitute(#5757FF, x$env)"
[37] "substitute(grey, x$env)"    "substitute(#8B8BFF, x$env)" "substitute(#3333FF, x$env)"
[40] "substitute(grey, x$env)"    "substitute(#FFFCFC, x$env)" "substitute(#FF7B7B, x$env)"
[43] "substitute(#FF9595, x$env)" "substitute(#F2F2FF, x$env)" "substitute(#FFE2E2, x$env)"
[46] "substitute(#CECEFF, x$env)" "substitute(#FF3D3D, x$env)" "substitute(#FF8B8B, x$env)"
[49] "substitute(grey, x$env)"    "substitute(grey, x$env)"    "substitute(#FFC8C8, x$env)"
[52] "substitute(#D8D8FF, x$env)" "substitute(#FF1414, x$env)" "substitute(#FF5757, x$env)"
[55] "substitute(#2424FF, x$env)" "substitute(#FF7171, x$env)" "substitute(#FF0000, x$env)"
[58] "substitute(#FF2E2E, x$env)" "substitute(#FF2424, x$env)" "substitute(#FF0A0A, x$env)"
```

This won't parse because it is not a legal R statement, but it is a valid color:
```
substitute(#2424FF, x$env)
```

Basically standard hex colors go into the parse statement as a comment character because for whatever reason that code strips the quotes off a vector of quoted colors =)

**Comment by hafen** [comment]
@jrounds did you resolve this?  If so, can you share what you learned?  If not, would it be easy to create an example?

**Comment by jrounds** [comment]
After I started simplifying it went away.  At the time it was in an old
ly_heatmap function I wrote, but I haven't had a chance to figure out if it
was a scope issue or what.  BTW when you wrap functions around rbokeh
functions in your own package these issues of arguments can get sticky -=)

Anyway in the end I had a working heatmap with color from ly_crect.

**Comment by hafen** [comment]
Oh no did you have issues even with 0.3.1?  Barret did a lot of work with lazy evaluation that should have cleared up issues with arguments inside functions, etc.

**Comment by jrounds** [comment]
Yeah I updated but this was a complex situation where a function called a
function in my own package environment which called ly_crect, so no telling
what actually the issue was.  If it comes up again I will track it down.
